src/dbnode/storage/index/types.go

```diff
@@ -94,6 +94,11 @@ type Results interface {
 	// including returning it to a backing pool.
 	Finalize()
 
+	// NoFinalize marks the Results such that a subsequent call to Finalize() will
+	// be a no-op and will not return the object to the pool or release any of its
+	// resources.
+	NoFinalize()
```

Comment:
this is only used in tests afaict, can you remove it?

Reply:
synced with Rob offline, since its already written and tested gonna leave it in so we can use for the query cache